### PR TITLE
Mbedtls link

### PR DESCRIPTION
**Pull request by disa6302**
*Issue #, if available:*

*What was changed?*
- Fixed the linking for LWS case to be crypto specific.

*Why was it changed?*
- The crypto linking is different for LWS vs CURL. In LWS case for this library, the library was always being linked to openssl, but it should link based on crypto library set up.

*How was it changed?*
- Linking to `PRODUCER_CRYPTO_LIBRARY` instead of Openssl only.

Before this change: (`libmbedcrypto.so.7 => not found`)

```
ubuntu@ip-172-31-11-82:~/amazon-kinesis-video-streams-producer-c/build$ ldd libkvsCommonLws.so
	linux-vdso.so.1 (0x00007fffcbf08000)
	libwebsockets.so.15 => /home/ubuntu/amazon-kinesis-video-streams-producer-c/open-source/lib/libwebsockets.so.15 (0x00007f32d5b7a000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f32d5800000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f32d5bcf000)
	libmbedtls.so.14 => not found
	libmbedx509.so.1 => not found
	libmbedcrypto.so.7 => not found
```

After this change: (`libmbedcrypto.so.7 => /home/ubuntu/amazon-kinesis-video-streams-producer-c/open-source/lib/libmbedcrypto.so.7 (0x00007f8518325000)`)

```
ubuntu@ip-172-31-11-82:~/amazon-kinesis-video-streams-producer-c/build$ ldd libkvsCommonLws.so
	linux-vdso.so.1 (0x00007ffdfdb56000)
	libmbedcrypto.so.7 => /home/ubuntu/amazon-kinesis-video-streams-producer-c/open-source/lib/libmbedcrypto.so.7 (0x00007f8518325000)
	libwebsockets.so.19 => /home/ubuntu/amazon-kinesis-video-streams-producer-c/open-source/lib/libwebsockets.so.19 (0x00007f85182c6000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f8518000000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f85183b4000)
	libmbedtls.so.14 => not found
	libmbedx509.so.1 => not found
```

*What testing was done for the changes?*
- Added CI to only test the build

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -289,8 +289,7 @@ if(BUILD_COMMON_LWS)
       set_target_properties(kvsCommonLws PROPERTIES VERSION ${KINESIS_VIDEO_PRODUCER_C_VERSION} SOVERSION ${KINESIS_VIDEO_PRODUCER_C_MAJOR_VERSION})
     endif()
     target_link_libraries(kvsCommonLws
-            ${OPENSSL_CRYPTO_LIBRARY}
-            ${OPENSSL_SSL_LIBRARY}
+            ${PRODUCER_CRYPTO_LIBRARY}
```
Comment (niyatim23) [inline CMakeLists.txt:292]: Shouldn't there be an if-else for openssl and mbedtls and ${OPENSSL_SSL_LIBRARY} removed only for mbedtls?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -176,6 +176,28 @@ jobs:
           cd build
           ./tst/producer_test
 
+  ubuntu-os-build-lws-mbedtls:
```
Comment (jdelapla) [inline .github/workflows/ci.yml:179]: add gcc to name

**Comment by hassanctech** [comment]
Nice!  Is it possible to print the results of `ldd` from before / after to confirm we're not linking to openssl any more, i.e. it's not specified anywhere in the `NEEDED` section of the ELF header?

File: CMakeLists.txt (hunk 1 above)
Comment (disa6302) [reply CMakeLists.txt:292]: https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/pull/442/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR263-R270

**Review by stefankiesz — APPROVED** [review]

**Comment by disa6302** [comment]
> Nice! Is it possible to print the results of `ldd` from before / after to confirm we're not linking to openssl any more, i.e. it's not specified anywhere in the `NEEDED` section of the ELF header?

Done!

**Review by niyatim23 — APPROVED** [review]

**Review by stefankiesz — APPROVED** [review]

**Comment by hassanctech** [comment]
> > Nice! Is it possible to print the results of `ldd` from before / after to confirm we're not linking to openssl any more, i.e. it's not specified anywhere in the `NEEDED` section of the ELF header?
> 
> Done!

Looking at the results it would say `not found` just because it can't find it in `LD_LIBRARY_PATH` or standard library path but in this case it looks like previously that library was just not built and put in the lib folder which is where it looks for it in the AFTER case. However what I was expecting to see in the BEFORE case is libssl/libcrypto in the list, since it is not there that would means previously it wasn't linking to openssl.

**Review by hassanctech — APPROVED** [review]
lgtm

**Pull request merged**